app/Gists/GistClient.php

```diff
@@ -76,4 +77,32 @@ public function postGistComment($gistId, $comment)
 
         return ResponseMediator::getContent($response);
     }
+
+    public function starGist($gistId)
+    {
+        $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
+        $this->github->getHttpClient()->put("https://api.github.com/gists/{$gistId}/star", json_encode(['body' => '']), ['Content-Length' => 0]);
+    }
+
+    public function unstarGist($gistId)
+    {
+        $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
+        $this->github->getHttpClient()->delete("https://api.github.com/gists/{$gistId}/star");
+    }
+
+    public function isStarredForUser($gistId)
+    {
+        if (Auth::check()) {
+            try {
+                $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
+                $this->github->getHttpClient()->get("https://api.github.com/gists/{$gistId}/star");
+
+                return true;
+            } catch (Throwable $e) {
+                return false;
+            }
+        }
+
+        return false;
```

Comment:
Can you do this, but also double-check that the implementation is working correctly? As-is it's expecting an exception to be thrown when you `get` the `/gists/gistId/star` endpoint, so we need to double-check that's what happens.

```suggestion
        if (Auth::guest()) {
            return false;
        }
        
        try {
           $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
           $this->github->getHttpClient()->get("https://api.github.com/gists/{$gistId}/star");

            return true;
        } catch (Throwable $e) {
            return false;
        }
```

Reply:
The actual API is [returning an HTTP status code](https://docs.github.com/en/free-pro-team@latest/rest/reference/gists#check-if-a-gist-is-starred) (204 if starred and 404 if not starred). 

The 404 is caught by the "catch" block, so the 'false' value is being returned. But, if I try to dump the actual exception, I'm getting an error message:  `Wrong parameters for Github\Exception\RuntimeException([string $message [, long $code [, Throwable $previous = NULL]]])`

Reply:
@faxblaster Huh. Would you drop the catch block for now so you can get a full stack trace?

Reply:
```
    public function isStarredForUser($gistId)
    {
        if (Auth::guest()) {
            return false;
        }

        $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
        $response = $this->github->getHttpClient()->get("https://api.github.com/gists/{$gistId}/star");

        return $response;
    }
```

**Response if Gist is unstarred for user**
![Screen Shot 2020-10-28 at 3 29 43 PM](https://user-images.githubusercontent.com/4378273/97493094-77914380-1932-11eb-989d-c8ba5e93724f.png)

Reply:
Huh. We'll have to pair some time on what exactly the first and second parameters are that are being passed here.